src/stores/VideoStore.js

```diff
@@ -310,6 +314,39 @@ class VideoStore {
 
     this.playoutOptions = playoutOptions;
   });
+
+  @action.bound
+  GenerateEmbedUrl = flow(function * ({objectId, versionHash}) {
+    let embedUrl = new URL("https://embed.v3.contentfabric.io");
+    const networkInfo = yield this.rootStore.client.NetworkInfo();
+    const networkName = networkInfo.name === "demov3" ? "demo" : (networkInfo.name === "test" && networkInfo.id === 955205) ? "testv4" : networkInfo.name;
+    const permission = yield this.rootStore.client.Permission({
+      objectId: objectId ? objectId : yield this.rootStore.client.utils.DecodeVersionHash(versionHash).objectId
+    });
+
+    embedUrl.searchParams.set("p", "");
+    embedUrl.searchParams.set("net", networkName);
+    embedUrl.searchParams.set("ct", "s");
+    embedUrl.searchParams.set("off", this.offering);
+
+    if(versionHash) {
+      embedUrl.searchParams.set("vid", versionHash);
+    } else if(objectId) {
+      embedUrl.searchParams.set("oid", objectId);
+    }
+
+    if(["owner", "editable", "viewable"].includes(permission)) {
+      const token = yield this.rootStore.client.CreateSignedToken({
+        objectId,
+        versionHash,
+        duration: 100 * 24 * 60 * 60 * 1000 // milliseconds
```

Comment:
do we really want to make a 100 day token?

Reply:
Apparently the fabric doesn't honor anything over 24 hours so I guess the value doesn't matter for now